src/datatype/svecf32.rs

```diff
@@ -0,0 +1,698 @@
+use crate::prelude::*;
+use pgrx::pg_sys::Datum;
+use pgrx::pg_sys::Oid;
+use pgrx::pgrx_sql_entity_graph::metadata::ArgumentError;
+use pgrx::pgrx_sql_entity_graph::metadata::Returns;
+use pgrx::pgrx_sql_entity_graph::metadata::ReturnsError;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlMapping;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlTranslatable;
+use pgrx::FromDatum;
+use pgrx::IntoDatum;
+use service::prelude::*;
+use std::alloc::Layout;
+use std::cmp::Ordering;
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::ptr::NonNull;
+
+#[repr(C, align(8))]
+pub struct SVecf32 {
+    varlena: u32,
+    len: u16,
+    kind: u8,
+    reserved: u8,
+    dims: u16,
+    phantom: [u8; 0],
```

Comment:
Add padding bits to force alignment to be 8.

Reply:
done in new commit